### ISSUE TITLE
Feature: Option for auto-generating defines for `#[cfg(…)]`

### ISSUE THREAD

**Issue by regexident** [eqrion/cbindgen#193]
The transformation step from `Cfg` to `Condition` introduced in https://github.com/eqrion/cbindgen/pull/191 levels the field for a second thing I'd like to add to cbindgen:

I'd like to add an option to the config: `missing-defines` with possible values `Warn` (implemented in this PR), `Ignore` (what we had before this PR) and `Generate`.

`Generate` would take `#[cfg(feature = "foo")]` and `#[cfg(test)]` and generate appropriate defines for them: `FEATURE_FOO` and `TEST` respectively.

A reasonable default would probably be `Warn`.

The idea behind `Generate` being that for most uses one would probably be fine with just auto-generated defines. No more having to add a `[defines]` section to `cbindgen.toml` and manually add/update entries whenever you add/change a `#[cfg(…)]`.

Manual entries in `[defines]` would act as overrides for auto-generated defines.

**Comment by regexident** [comment]
I'd be happy to implement this and make a PR if you'd be okay with adding such a feature.

One would probably want to add the config's prefix to auto-generated defines to prevent collisions (`TEST` is highly likely to exist elsewhere in a C code-base, e.g.).

**Comment by eqrion** [comment]
Yeah I like the idea behind this. The original implementation of #[cfg] was done quick and conservative. This would help make it easier to use for most cases.

**Comment by regexident** [comment]
Cool. I'll work on it once https://github.com/eqrion/cbindgen/pull/191 is merged then (as things would depend on it)! 👍🏻
